File: consonant_reversal.java
import java.io.*;
import java.util.*;

public class Solution {
    private static boolean isVower(char c){
        if ( c=='a' || c=='e' || c=='i' || c=='o' || c=='u')
            return true;
        else
            return false;
    }
    private static void reverseString(String str){
        int len = str.length();
        char cur[] = str.toCharArray();
        int left = 0;
        int right = len-1;
        while(left<right){
            if ( isVower(cur[left]))
                left++;
            else if (isVower(cur[right]))
                right--;
            else
            {
                char c = cur[left];
                cur[left] = cur[right];
                cur[right] = c;
                left++;
                right--;
            }
        }
        System.out.println(new String(cur));
    }
    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        try {
           BufferedReader in = new BufferedReader( new InputStreamReader(System.in));
           int n = Integer.parseInt( in.readLine().trim());
        
           for(int i=0; i<n; ++i)
           {
               String str = in.readLine().trim();
               reverseString(str);
           }
        
           in.close();
        }
        catch(IOException e){
            e.printStackTrace();
        }
    }
}

File: consonant_reversal.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <string>
using namespace std;

string reverseVowel(const string& str){
    int len = str.length();
    string res(len, '*');
    string tmp = "";
    for(int i=0; i<len; ++i)
    {
        if (str[i]=='a' || str[i]=='e' || str[i]=='i' || str[i]=='o' || str[i]=='u')
            res[i] = str[i];
        else
            tmp += str[i];
    }
    
    int i=0;
    int j=tmp.length()-1;
    while(i<j)
    {
        char c = tmp[i];
        tmp[i] = tmp[j];
        tmp[j] = c;
        i++;
        j--;
    }
    
    int k=0;
    for(int i=0; i<len; ++i)
    {
        if (res[i]=='*')
           res[i] = tmp[k++];  
    }
    
    return res;
}
int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */ 
    int num = 0;
    cin >> num;
    for(int i=0; i<num; ++i)
    {
        string input;
        cin >> input;
        cout << reverseVowel(input) << "\n";    
    }
    
    return 0;
}

